packages/realm-server/migrations/1728561353454_create-user-billing.js

```diff
@@ -0,0 +1,164 @@
+exports.up = (pgm) => {
+  pgm.createTable('users', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    matrix_user_id: { type: 'varchar', notNull: true, unique: true },
+    stripe_customer_id: { type: 'varchar' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('plans', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    name: { type: 'varchar', notNull: true },
+    monthly_price: { type: 'numeric(19,4)', notNull: true },
+    credits_included: { type: 'integer', notNull: true },
+    overage_fee_per_credit_in_usd: { type: 'numeric(19,4)' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('billing_cycles', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    subscription_id: { type: 'integer' },
+    period_start: { type: 'timestamp', notNull: true },
+    period_end: { type: 'timestamp', notNull: true },
+  });
+
+  pgm.createTable('subscriptions', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    user_id: { type: 'integer', references: 'users(id)' },
+    plan_id: { type: 'integer', references: 'plans(id)' },
+    started_at: { type: 'timestamp', notNull: true },
+    ended_at: { type: 'timestamp', notNull: true },
+    status: { type: 'varchar', notNull: true },
+    stripe_subscription_id: { type: 'varchar', notNull: true },
+    current_billing_cycle_id: {
+      type: 'integer',
+      references: 'billing_cycles(id)',
+    },
+  });
+
+  pgm.addConstraint('billing_cycles', 'billing_cycles_subscription_id_fkey', {
+    foreignKeys: {
+      columns: 'subscription_id',
+      references: 'subscriptions(id)',
+    },
+  });
+
+  pgm.createTable('ai_actions', {
+    id: {
+      type: 'serial',
+      primaryKey: true,
+    },
+    action_type: { type: 'varchar', notNull: true },
+    user_id: { type: 'integer', references: 'users(id)' },
+    cost_in_usd: { type: 'numeric(19,4)', notNull: true },
+    credits_used: { type: 'numeric(19,4)', notNull: true },
+    is_overage: { type: 'boolean', notNull: true, default: false },
```

Comment:
AI bot will need to figure out whether plan credits were already used, and in that case flag the AI action as "overage"